test/strategy/auth0/oauth_test.exs

```diff
@@ -21,7 +22,8 @@ defmodule Ueberauth.Strategy.Auth0.OAuthTest do
 
   test "raises when there is no configuration" do
     assert_raise(RuntimeError, ~r/^Expected to find settings under.*/, fn ->
-      client(otp_app: :unknown_auth0_otp_app)
+      conn = %Plug.Conn{}
+      client(conn, otp_app: :unknown_auth0_otp_app)
```

Comment:
I'd add a few tests here to check that the settings are fetched from the `ConfigFrom` module as expected.

Reply:
I'm trying to get rid of the `mix test.all` alias to just use `mix test`.

I've added some tests to `oauth_test.exs` to check that the `ConfigFrom` module was set as expected. Unfortunately, they're not working and I believe it's because the configurations are being wrongly loaded on **runtime** (see ["How to dynamically load configurations for tests at runtime within Elixir libraries"](https://dev.to/alexjuca/how-to-dynamically-load-configurations-for-tests-at-runtime-within-elixir-libraries-2ign)).

Do you think it's necessary for this case to create multiple environments to test the possible configurations?

Another references:
- [Separate mix commands for unit and integration tests](https://elixirforum.com/t/separate-mix-commands-for-unit-and-integration-tests/32370)
- [Elixir test multiple environments](https://spin.atomicobject.com/2018/10/22/elixir-test-multiple-environments/)

Reply:
I forgot to mention that `Mix.Config` is **deprecated**: https://hexdocs.pm/mix/Mix.Config.html#read!/2

Reply:
@achedeuzot I'm not sure if you saw this comment before commented "LGTM 🎉 " at the bottom 😬

Reply:
Ah no sorry, I commented LGTM before running the CI tests. I see the errors. 

If it works with the `mix test.all`, I'm okay with it until we may find a better way :)

Reply:
I've changed the deprecated `Mix.Config` with `Config`. Also, I continued working on the `OAuthTest` module but I couldn't make the tests related to `config_from` pass. I don't know why the config loading from the test `setup` is not working. I've also tried creating multiple `mix aliases` on another branch, hoping that if the App is being recompiled each time with a different environment but it also failed.

Can you take a look to the code? Maybe you can find something that my eyes are not seeing.

Cheers.

Reply:
Hey @silvagustin,

Thanks so much for your efforts to make the tests work, really appreciate it ! Just a quick message to let you know that I'll be having a lot on my plate in the upcoming days and won't be available before at least next Tuesday (18th of may) to look into it. I'll look at it then.

Otherwise, I'm okay if you want to revert to one of your other solutions that seemed to work (with the 2 environments in mix.exs) if you want to move forward with this PR.

Cheers,